### PR TITLE
[CALCITE-6258] Map value constructor is unparsed incorrectly for PrestoSqlDialect

### PR DESCRIPTION
**Pull request by YiwenWu**
fixed Presto Map Function is unparsed incorrectly ：map(array<K>, array<V>) 
https://issues.apache.org/jira/browse/CALCITE-6258

### CODE REVIEW

**Comment by JiajunBernoulli** [comment]
Good catch. 
I have a similar PR: https://github.com/linkedin/coral/pull/99/files

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -4169,7 +4171,10 @@ private SqlDialect nonOrdinalDialect() {
     final String query = "SELECT MAP['k1', 'v1', 'k2', 'v2']";
     final String expected = "SELECT MAP ('k1', 'v1', 'k2', 'v2')\n"
         + "FROM (VALUES (0)) t (ZERO)";
-    sql(query).withSpark().ok(expected);
+    final String expectedPresto = "SELECT MAP (ARRAY['k1', 'k2'], ARRAY['v1', 'v2'])\n"
```
Comment (caicancai) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:4174]: Maybe we should add a comment with a jira link in the comment

File: core/src/main/java/org/apache/calcite/sql/dialect/PrestoSqlDialect.java [hunk 2]
```diff
@@ -131,8 +136,30 @@ private static void unparseUsingLimit(SqlWriter writer, @Nullable SqlNode offset
       RelToSqlConverterUtil.specialOperatorByName("APPROX_DISTINCT")
           .unparse(writer, call, 0, 0);
     } else {
-      // Current impl is same with Postgresql.
-      PostgresqlSqlDialect.DEFAULT.unparseCall(writer, call, leftPrec, rightPrec);
+      switch (call.getKind()) {
```
Comment (JiajunBernoulli) [inline core/src/main/java/org/apache/calcite/sql/dialect/PrestoSqlDialect.java:139]: Extract them to `createMapValueConstuctor` function is bettter.
> I like single responsibility principle

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (YiwenWu) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: done, added new tests

File: core/src/main/java/org/apache/calcite/sql/dialect/PrestoSqlDialect.java (hunk 2 above)
Comment (YiwenWu) [reply core/src/main/java/org/apache/calcite/sql/dialect/PrestoSqlDialect.java:143]: done, extract new function `convertMapValueCall`

Comment (JiajunBernoulli) [reply core/src/main/java/org/apache/calcite/sql/dialect/PrestoSqlDialect.java:143]: Why don't we extract more thoroughly?

We can use more parameter like `unparseCall(writer, call, leftPrec, rightPrec)`

Comment (YiwenWu) [reply core/src/main/java/org/apache/calcite/sql/dialect/PrestoSqlDialect.java:143]: done, extract map unparse  method `unparseMapValue`

**Review by macroguo-ghy — APPROVED** [review]
+1 for this pr.

But `MAP[]` will be rejected by `SqlMapValueConstructor#checkOperandTypes`, so I think we need add a new `SqlLibrary` and add (or enable) `map` function for presto.

**Comment by macroguo-ghy** [comment]
Please squash your commits, I will merge it later. @YiwenWu

**Pull request merged**